src/resources/extensions/quarto/video/video.lua

```diff
@@ -60,16 +71,16 @@ local replaceCommonAttributes = function(snippet, params)
   return result
 end
 
-local checkEndMatch = function(value, matcherFront)
+local checkMatchStart = function(value, matcherFront)
   return string.match(value, matcherFront .. '(.-)$')
 end
```

Comment:
I understand why you changed name, but I would have expect a function that check on start of the string to use a regex starting with special char `^` which means start of string match. 

I believe you function would also match something like `anythinghttps://vimeo.com/`

Probably won't happen, but I am curious why match on anything on the end using `(.-)$`. 

maybe there is the reason I don't see. 

Everythings works - so probably no problem but usually using `^` for matching at start of string.

Reply:
Thanks for suggesting this.  Can you help me with a unit test for this?  I'm not sure how I can get it to fail with a unit test.

Reply:
I am not sure how unit test works in Lua but I would suggest either 

* to test `checkMatchStart()` according to what you expect as results - if you expect to be a string that "start with match" , I think you won't get this with the regex used. (from quick test in repl.it with Lua)

* or directly your video function where I would expect `{{ video anythinghttps://vimeo.com/ }}` to be ignore, or shown an invalid URL. 

Are you testing already in the code path for the URL part of the video shortcode to be either a URL or a local file ? I believe we do that elsewhere in our TS code base. 

Maybe it is too much to check, and we can assume the user would use a valid url starting with `http(s)`